src/lib/indexer/__tests__/fixtures/db.ts

```diff
@@ -0,0 +1,44 @@
+import sinon from 'sinon'
+import Database from 'src/lib/db'
+
+type LastProcessBlockResult = { hash: string; parent: string; height: number } | null
+
+export const withLastProcessedBlocksByCall = (calls: LastProcessBlockResult[]) => {
+  let getMock = sinon.stub().resolves(calls.at(-1))
+  for (let i = 0; i < calls.length; i++) {
+    getMock = getMock.onCall(i).resolves(calls[i])
+  }
+
+  const insertProcessedBlock = sinon.stub().resolves()
+
+  const self = {
+    getLastProcessedBlock: getMock,
+    withTransaction: sinon.spy(async function (fn: (db: Database) => Promise<void>) {
+      await fn({
+        insertProcessedBlock,
+      } as unknown as Database)
+    }),
+  } as unknown as Database
+
+  return self
+}
+
+export const withInitialLastProcessedBlock = (initial: LastProcessBlockResult) => {
+  let lastBlock: LastProcessBlockResult = initial
+  const getMock = sinon.spy(() => Promise.resolve(lastBlock))
+
+  const insertProcessedBlock = sinon.spy((block: LastProcessBlockResult) => {
+    lastBlock = block
+    return Promise.resolve()
+  })
+
+  return {
+    getLastProcessedBlock: getMock,
+    insertProcessedBlock,
+    withTransaction: sinon.spy(async function (fn: (db: Database) => Promise<void>) {
+      await fn({
+        insertProcessedBlock,
+      } as unknown as Database)
+    }),
+  } as unknown as Database
+}
```

Comment:
It's worth saying that I'm torn on whether this is a good mocking approach in this instance. This is essentially a working mock and so I'm relying on the behaviour of the working mock against other assertions rather than asserting directly that the db was updated correctly. Not ideal but the mock is very simple so I think it's probably ok